src/build.sh

```diff
@@ -81,23 +83,29 @@ function kernel_build()
     exit 125 # ECANCELED
   fi
 
-  if [ -x "$(command -v nproc)" ]; then
-    parallel_cores=$(nproc --all)
+  if command_exists nproc; then
+    parallel_cores="$(nproc --all)"
   else
-    parallel_cores=$(grep -c ^processor /proc/cpuinfo)
+    parallel_cores="$(grep -c ^processor /proc/cpuinfo)"
+  fi
+
+  optimizations="-j$((parallel_cores * cpu_scaling_factor / 100))"
+
+  if [[ -n "${options_values['CCACHE']}" ]]; then
+    optimizations="CC=\"ccache gcc -fdiagnostics-color\" $optimizations"
   fi
 
   if [[ -n "$doc_type" ]]; then
-    command="make -j$parallel_cores $doc_type"
+    command="make $optimizations $doc_type"
     cmd_manager "$flag" "$command"
     return
   fi
 
+  command="make $optimizations ARCH=$arch $cross_compile"
```

Comment:
https://kworkflow.org/content/codingstyle.html#string-concatenation
(same for all commits)